File: Python/Coffee Machine/coffee_machine.py

def check_resources(resources, coffee):
    for i in range(len(resources)):
        if resources[i] - coffee[i] < 0:
            if i == 0:
                print("Sorry, not enough water!")
            elif i == 1:
                print("Sorry, not enough milk!")
            elif i == 2:
                print("Sorry, not enough coffee beans!")
            elif i == 3:
                print("Sorry, not enough disposable cups!")
            return False
    return True

def buy(resources, type):
    print("What do you want to buy? 1 - espresso, 2 - latte, 3 - cappuccino, back - to main menu:")
    choice = input()
    if choice == "back":
        return resources

    if check_resources(resources, type[int(choice) - 1]):
        print("I have enough resources, making you a coffee!")
    else:
        return resources

    if choice == "1":
        for i in range(len(resources)):
            resources[i] -= type[0][i]
    elif choice == "2":
        for i in range(len(resources)):
            resources[i] -= type[1][i]
    elif choice == "3":
        for i in range(len(resources)):
            resources[i] -= type[2][i]
    return resources

def fill(resources):
    print("Write how many ml of water you want to add:")
    resources[0] += int(input())
    print("Write how many ml of milk you want to add: ")
    resources[1] += int(input())
    print("Write how many grams of coffee beans you want to add:")
    resources[2] += int(input())
    print("Write how many disposable cups of coffee you want to add: ")
    resources[3] += int(input())
    return resources

def take(resources):
    print("I gave you $" + str(resources[4]))
    resources[4] = 0
    return resources

def remaining(resources):
    print("The coffee machine has:")
    print(resources[0], "ml of water")
    print(resources[1], "ml of milk")
    print(resources[2], "g of coffee beans")
    print(resources[3], "disposable cups")
    print("$" + str(resources[4]), "of money")

def main():
    espresso = [250, 0, 16, 1, -4]
    latte = [350, 75, 20, 1, -7]
    cappuccino = [200, 100, 12, 1, -6]
    type = [espresso, latte, cappuccino]
    resources = [400, 540, 120, 9, 550]
    while True:
        print("Write action (buy, fill, take, remaining, exit):")
        action = input()
        if action == "buy":
            resources = buy(resources, type)
        elif action == "fill":
            resources = fill(resources)
        elif action == "take":
            resources = take(resources)
        elif action == "remaining":
            remaining(resources)
        elif action == "exit":
            exit(1)

main()

File: Java/Coffee Machine/src/machine/CoffeeMachine.java
package machine;

import java.util.Scanner;

public class CoffeeMachine {

    public static int[] buy(int[] res) {
        Scanner scan = new Scanner(System.in);
        System.out.println("What do you want to buy? 1 - espresso, 2 - latte, 3 - cappuccino, back - to main menu:");
        String choice = scan.next();
        if (choice.equals("back"))
            return res;
        switch (choice) {
            case "1":
                if (res[0] > 249 && res[2] > 15 && res[3] > 0) {
                    System.out.println("I have enough resources, making you a coffee!");
                    res[0] -= 250;
                    res[2] -= 16;
                    res[3]--;
                    res[4] += 4;
                } else {
                    if (res[0] < 250)
                        System.out.println("Sorry, not enough water!");
                    if (res[2] < 16)
                        System.out.println("Sorry, not enough coffee beans!");
                    if (res[3] < 1)
                        System.out.println("Sorry, not enough disposable cups!");
                }
                break;
            case "2":
                if (res[0] > 349 && res[1] > 74 && res[2] > 19 && res[3] > 0) {
                    System.out.println("I have enough resources, making you a coffee!");
                    res[0] -= 350;
                    res[1] -= 75;
                    res[2] -= 20;
                    res[3]--;
                    res[4] += 7;
                } else {
                    if (res[0] < 350)
                        System.out.println("Sorry, not enough water!");
                    if (res[1] < 75)
                        System.out.println("Sorry, not enough milk!");
                    if (res[2] < 20)
                        System.out.println("Sorry, not enough coffee beans!");
                    if (res[3] < 1)
                        System.out.println("Sorry, not enough disposable cups!");
                }
                break;
            case "3":
                if (res[0] > 199 && res[1] > 99 && res[2] > 11 && res[3] > 0) {
                    System.out.println("I have enough resources, making you a coffee!");
                    res[0] -= 200;
                    res[1] -= 100;
                    res[2] -= 12;
                    res[3]--;
                    res[4] += 6;
                } else {
                    if (res[0] < 200)
                        System.out.println("Sorry, not enough water!");
                    if (res[1] < 100)
                        System.out.println("Sorry, not enough milk!");
                    if (res[2] < 12)
                        System.out.println("Sorry, not enough coffee beans!");
                    if (res[3] < 1)
                        System.out.println("Sorry, not enough disposable cups!");
                }
        }
        return res;
    }

    static int[] fill (int[] res) {
        Scanner scan = new Scanner(System.in);
        System.out.println("Write how many ml of water you want to add:");
        res[0] += scan.nextInt();
        System.out.println("Write how many ml of milk you want to add:");
        res[1] += scan.nextInt();
        System.out.println("Write how many grams of coffee beans you want to add:");
        res[2] += scan.nextInt();
        System.out.println("Write how many disposable cups of coffee you want to add:");
        res[3] += scan.nextInt();
        return res;
    }

    static int[] take (int[] res) {
        System.out.printf("I gave you $%d\n", res[4]);
        res[4] = 0;
        return res;
    }

    static void remaining (int[] res) {
        System.out.println("The coffee machine has:");
        System.out.printf("%d ml of water\n", res[0]);
        System.out.printf("%d ml of milk\n", res[1]);
        System.out.printf("%d g of coffee beans\n", res[2]);
        System.out.printf("%d disposable cups\n", res[3]);
        System.out.printf("$%d of money\n", res[4]);
    }

    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        int[] res = {400, 540, 120, 9, 550};
        while (true) {
            System.out.println("\nWrite action (buy, fill, take, remaining, exit):");
            String choice = scan.next();

            if (choice.equals("exit")) {
                System.out.println("Bye!");
                break;
            } else if (choice.equals("buy"))
                res = buy(res);
            else if (choice.equals("fill"))
                res = fill(res);
            else if (choice.equals("take"))
                res = take(res);
            else if (choice.equals("remaining"))
                remaining(res);
            else
                System.out.println("Incorrect action!");

        }
    }
}

